tests/tck/features/optimizer/PushSampleDownRule.feature

```diff
@@ -0,0 +1,79 @@
+      | 12 | Limit        | 13           |                                                        |
+      | 13 | GetNeighbors | 6            | {"limit": "2", "random": "false"}                      |
+      | 6  | Loop         | 0            | {"loopBody": "5"}                                      |
+      | 5  | Dedup        | 4            |                                                        |
+      | 4  | Project      | 20           |                                                        |
+      | 20 | Limit        | 21           |                                                        |
+      | 21 | GetNeighbors | 1            | {"limit": "$__VAR_2[($__VAR_1-1)]", "random": "false"} |
+      | 1  | Start        |              |                                                        |
+      | 0  | Start        |              |                                                        |
+
+  Scenario: push sample down to GetNeighbors
+    When profiling query:
+      """
+      GO 1 STEPS FROM "James Harden" OVER like REVERSELY SAMPLE [2]
+      """
+    Then the result should be, in any order:
+      | like._dst |
+      | /[\w\s]+/ |
+      | /[\w\s]+/ |
+    And the execution plan should be:
+      | id | name         | dependencies | operator info                    |
+      | 1  | Project      | 2            |                                  |
+      | 2  | Sample       | 3            |                                  |
+      | 3  | GetNeighbors | 4            | {"limit": "2", "random": "true"} |
+      | 4  | Start        |              |                                  |
+    When profiling query:
+      """
+      GO 2 STEPS FROM "James Harden" OVER like REVERSELY SAMPLE [2, 2]
+      """
+    Then the result should be, in any order:
+      | like._dst |
+      | /[\w\s]+/ |
+      | /[\w\s]+/ |
+    And the execution plan should be:
+      | id | name         | dependencies | operator info                                         |
+      | 9  | Project      | 12           |                                                       |
+      | 12 | Sample       | 13           |                                                       |
+      | 13 | GetNeighbors | 6            | {"limit": "2", "random": "true"}                      |
+      | 6  | Loop         | 0            | {"loopBody": "5"}                                     |
+      | 5  | Dedup        | 4            |                                                       |
+      | 4  | Project      | 20           |                                                       |
+      | 20 | Sample       | 21           |                                                       |
+      | 21 | GetNeighbors | 1            | {"limit": "$__VAR_2[($__VAR_1-1)]", "random": "true"} |
+      | 1  | Start        |              |                                                       |
+      | 0  | Start        |              |                                                       |
```

Comment:
You could add some operator info to illustrate `"limit": "$__VAR_2[($__VAR_1-1)]"`, like where does the `$__VAR_2` and `($__VAR_1-1)` come from.

Reply:
It's just inner variable.

Reply:
I get it.